draft-ietf-privacypass-integration.md

```diff
@@ -0,0 +1,147 @@
+- Blinding/unblinding: A pair of cryptographic operations that allow
+  randomly distributing and revealing some underlying `data`, whilst
+  preserving the ability to generate valid signatures on `data` itself.
+
+We assume that all protocol messages are encoded into raw byte format
+before being sent across the wire.
+
+# Privacy Pass flow {#protocol-flow}
+
+There are three phases in the protocol: the initialization phase, the
+issuance phase, and the redemption phase.
+
+In the initialization phase, the server samples a keypair and publishes
+a ``public key`` (or key commitment) to a public repository.
+
+In the issuance phase:
+
+- The client and server optionally agree on some public ``metadata``.
+- The client retrieves an servers public key, and generates some initial
+  ``token`` data. The client cryptographically ``blinds`` this data, and
+  sends it to the server.
+- The server ``signs`` the blinded token data (including optional), and
+  produces a proof that it used the committed keypair, and sends the
+  signature and proof back to the client.
+- The client ``verifies`` the proof, ``unblinds`` the signature, and
+  stores an authenticated triple of the token, optional metadata, and
+  unblinded signature.
+
+In the redemption phase:
+
+- The client retrieves an authenticated token, optional metadata and
+  signature triple, and sends it to the server.
+- The server ``validates`` that the pair is authenticated correctly and
+  authorizes the client.
+
+# Partially Oblivious Pseudorandom Function Protocol
+
+We can instantiate the protocol flow in {{protocol-flow}} using the
+partially oblivious pseudorandom function (POPRF) protocol in
+{{I-D.irtf-cfrg-voprf}}. In summary, the issuance phase corresponds to
+receiving a pseudorandom function evaluation on the blinded data (with
+optional metadata). The redemption phase corresponds to revealing
+`finalized` data back to the original issuing server.
+
+Note that this instantiation only provides a symmetric verification
+mechanism, since the verification of redemptions can only be performed
+by the server possessing the secret issuing key. In
+{{I-D.ietf-privacypass-architecture}}, we provide alternative frameworks
+for allowing asynchronous and delegated verification of tokens.
+
+## Security guarantees
+
+The privacy of clients is determined by the unlinkability of client
+requests during the POPRF protocol. Moreover, the one-more-forgery
+security of the POPRF prevents clients from forging valid tokens for a
+given server. See {{I-D.irtf-cfrg-voprf}} for more details.
+
+## Metadata
+
+The POPRF protocol provides mechanisms for embedding public metadata
+into the function evaluations. Such metadata should be agreed apriori by
+clients and servers, and is regarded as being public to entities that
+are not even included in the explicit issuance and redemption exchanges.
+
+# Protocol ciphersuites {#ciphersuites}
+
+Ciphersuite negotiation is only relevant in terms of negotiating the
```

Comment:
Negotiation is a bit odd here, since nothing's negotiated. Maybe this should say that ciphersuites are determined by the server's choice of configuration?